### DISCUSSION TITLE
Questions on route() helper

### DISCUSSION THREAD

**Discussion by StefMTP** [adonisjs/core#2978] [Help] [answered]
I am trying to use named routes for my app but I am having some conflicts with multiple query string parameters. Let's suppose that this is the code I have right now:
```
  {{
    route('route_name', {
                qs: {
                    shop: nameOfShop,
                    xml_variation: nameOfVariation
                }
            })
    }}
```
For some reason, this translates to **'/routeurl?shop=nameOfShop'&amp';xml_variation=nameOfVariation'**.
Why am I getting a second ampersand in there?
Also, I am getting the following warning:
**'(node:9447) You are using legacy the API of the "Route.makeUrl". We recommend reading the docs and use the latest API: DeprecationWarning'**

**Comment by thetutlage** [comment, marked as answer]
Just tried it and seems to be working fine. 

```
      <a href="{{ route('/', { qs: { shop: 'bar', xml_variation: 'foo' } }) }}">
        {{ route('/', { qs: { shop: 'bar', xml_variation: 'foo' } }) }}
      </a>
```

<img width="489" alt="Screenshot 2021-07-29 at 10 36 37 AM" src="https://user-images.githubusercontent.com/1706381/127435011-c68c8708-a598-4dc3-878d-4fbb8c5dc8b8.png">

Regarding deprecation. It's because you are using the legacy syntax. If you have spotted that in the docs, then do lemme know, otherwise just follow the docs :)

**Reply by StefMTP** [reply]
The problem with the second ambersand seems to appear when I am trying to wrap the edge expression in "", so I can pass it as the url in a fetch expression.
I am not using the makeUrl method on my own, should I? I thought the route helper basically translates into a route generator. I found all that in the reference for URL generation https://docs.adonisjs.com/guides/routing#url-generation

**Reply by thetutlage** [reply]
You need to share the actual code snippets

**Reply by thetutlage** [reply]
The docs mention the query string as the 3rd param and not 2nd. https://docs.adonisjs.com/guides/routing#append-query-string

Also check this https://docs.adonisjs.com/reference/views/globals/route#query-string

**Reply by StefMTP** [reply]
I think I solved it now, what I was doing was passing this: '{{ route("route_name", [], { qs: { xml_variation: nameOfVariation, shop: nameOfShop } }) }}' as the url parameter in a fetch. What was actually happening, was that the & was being translated into amp;, ruining the query string. By wrapping the expression into three curly braces, I managed to fix it. So I am assumming that in the case that no params are required, passing in an empty array is fine?

**Reply by thetutlage** [reply]
> So I am assumming that in the case that no params are required, passing in an empty array is fine?

Yup
